go-controller/pkg/ovn/gress_policy.go

```diff
@@ -97,6 +97,28 @@ func (gp *gressPolicy) ensurePeerAddressSet(factory addressset.AddressSetFactory
 	return nil
 }
 
+func (gp *gressPolicy) addPeerSvcVip(service *v1.Service) error {
+	if gp.peerAddressSet == nil {
+		return fmt.Errorf("peer AddressSet is nil, cannot add peer Service: %s for gressPolicy: %s",
+			service.ObjectMeta.Name, gp.policyName)
+	}
+	ip := getSvcVips(service)
+
+	klog.V(5).Infof("Adding SVC clusterIP to gressPolicy's Address Set: %v", ip)
+	return gp.peerAddressSet.AddIPs(ip)
```

Comment:
if ips is nil/empty slice here, AddIPs will work, but it will lock for no reason. Maybe add an extra commit here to check in AddIPs if the slice is empty and then return, before trying to lock the address set.